### PR TITLE
Network manager

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: include/aos/sm/networkmanager.hpp [hunk 1]
```diff
@@ -360,6 +360,60 @@ class TrafficMonitorItf {
...
+    virtual Error CreateNetworkNamespace(const String& instanceID) const = 0;
+
+    /**
+     * Returns network namespace path.
+     *
+     * @param instanceID instance ID.
+     * @param[out] netnsPath network namespace path.
+     * @return Error.
+     */
+    virtual Error GetNetworkNamespacePath(const String& instanceID, String& netnsPath) const = 0;
```
Comment (al1img) [inline include/aos/sm/networkmanager.hpp:387]: Can we have 'RetwithError<StaticString<cFilePathLen>>'? We use it for string value a lot as I see.

File: tests/include/mocks/namespaceprovidermock.hpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#include <gmock/gmock.h>
+
+#include "aos/sm/networkmanager.hpp"
+
+namespace aos::sm::networkmanager {
+
+/**
+ * Namespace provider mock.
+ */
+class NamespaceProviderMock : public NamespaceProviderItf {
```
Comment (al1img) [inline tests/include/mocks/namespaceprovidermock.hpp:19]: Put into networkmanagermock.hpp keep mocks in the corresponding to original header files.

File: tests/include/mocks/netinterfaceprovidermock.hpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+#include <gmock/gmock.h>
+
+#include "aos/sm/networkmanager.hpp"
+
+namespace aos::sm::networkmanager {
+
+/**
+ * Network interface provider interface.
+ */
+class NetworkInterfaceProviderMock : public NetworkInterfaceProviderItf {
```
Comment (al1img) [inline tests/include/mocks/netinterfaceprovidermock.hpp:19]: Ditto

File: include/aos/sm/networkmanager.hpp [hunk 4]
```diff
@@ -360,6 +359,59 @@ class TrafficMonitorItf {
         = 0;
 };
 
+/**
+ * Namespace provider interface.
+ */
+class NamespaceProviderItf {
```
Comment (al1img) [inline include/aos/sm/networkmanager.hpp:365]: Better to name it Manager or handler.

File: include/aos/sm/networkmanager.hpp [hunk 5]
```diff
@@ -360,6 +359,59 @@ class TrafficMonitorItf {
...
+     * @param instanceID instance ID.
+     * @return Error.
+     */
+    virtual Error DeleteNetworkNamespace(const String& instanceID) const = 0;
+};
+
+/**
+ * Network interface provider interface.
+ */
+class NetworkInterfaceProviderItf {
```
Comment (al1img) [inline include/aos/sm/networkmanager.hpp:399]: ditto

**Review by mlohvynenko — APPROVED** [review]

File: include/aos/sm/networkmanager.hpp [hunk 6]
```diff
@@ -360,6 +359,59 @@ class TrafficMonitorItf {
...
+     * Destructor.
+     */
+    virtual ~NamespaceProviderItf() = default;
+
+    /**
+     * Creates network namespace.
+     * @param instanceID instance ID.
+     * @return Error.
+     */
+    virtual Error CreateNetworkNamespace(const String& instanceID) const = 0;
```
Comment (al1img) [inline include/aos/sm/networkmanager.hpp:377]: Looks like it can't be const method.

File: include/aos/sm/networkmanager.hpp [hunk 7]
```diff
@@ -360,6 +359,59 @@ class TrafficMonitorItf {
...
+     */
+    virtual RetWithError<StaticString<cFilePathLen>> GetNetworkNamespacePath(const String& instanceID) const = 0;
+
+    /**
+     * Deletes network namespace.
+     *
+     * @param instanceID instance ID.
+     * @return Error.
+     */
+    virtual Error DeleteNetworkNamespace(const String& instanceID) const = 0;
```
Comment (al1img) [inline include/aos/sm/networkmanager.hpp:393]: not const

File: include/aos/sm/networkmanager.hpp [hunk 8]
```diff
@@ -360,6 +359,59 @@ class TrafficMonitorItf {
...
+     */
+    virtual ~NetworkInterfaceProviderItf() = default;
+
+    /**
+     * Removes interface.
+     *
+     * @param ifname interface name.
+     * @return Error.
+     */
+    virtual Error RemoveInterface(const String& ifname) const = 0;
```
Comment (al1img) [inline include/aos/sm/networkmanager.hpp:412]: not const

**Review by al1img — APPROVED** [review]

**Pull request merged**